Assets/Scripts/EnemySpawn.cs

```diff
@@ -3,30 +3,108 @@
 using UnityEngine;
 using UnityEngine.AI;
 
+[System.Serializable]
+public class Wave
+{
+    public string waveName;
+    public int numOfEnemies;
+    public string[] typeOfEnemies;
+    public float spawnInterval;
+
+    public Wave(string name, int num, string[] types, float interval)
+    {
+        waveName = name;
+        numOfEnemies = num;
+        typeOfEnemies = types;
+        spawnInterval = interval;
+    }
+}
+
 public class EnemySpawn : MonoBehaviour
 {
     private Vector3 _enemySpawn = Vector3.zero;
     private bool _cooldown = false;
     private float _cooldownDelay = SpeedManager.enemySpawnScaling;
     private const float platformRadius = 175/2;
 
+    private bool canSpawn = false;
+    private bool startSpawn = false;
+    private float nextSpawnTime;
+    private Wave currentWave;
+    private int currentWaveNumber = 0;
+
+    private const int totalWaveNumber = 3;
+    private const int enemiesPerWave = 10;
+    private const int spawnInterval = 0;
+    private int currentNumEnemies = 0;

```

Comment:
dont have to put `private`, variables are by default private

Reply:
thank you max